NuRadioMC/simulation/simulation.py

```diff
@@ -147,112 +1094,103 @@ def __init__(self, inputfilename,
             # random seed once and save this seed to the config setting. If the simulation is rerun, we can get
             # the same random sequence.
-            self._cfg['seed'] = np.random.randint(0, 2 ** 32 - 1)
+            self._config['seed'] = np.random.randint(0, 2 ** 32 - 1)
 
-        self._rnd = Generator(Philox(self._cfg['seed']))
+        self._rnd = Generator(Philox(self._config['seed']))
 
         self._outputfilename = outputfilename
         if os.path.exists(self._outputfilename):
             msg = f"hdf5 output file {self._outputfilename} already exists"
-            if file_overwrite == False:
+            if file_overwrite is False:
                 logger.error(msg)
                 raise FileExistsError(msg)
-            else:
-                logger.warning(msg)
-        self._detectorfile = detectorfile
-        self._n_reflections = int(self._cfg['propagation']['n_reflections'])
+            logger.warning(msg)
+
         self._outputfilenameNuRadioReco = outputfilenameNuRadioReco
-        self._debug = debug
         self._evt_time = evt_time
         self.__write_detector = write_detector
         logger.status("setting event time to {}".format(evt_time))
         self._event_group_list = event_list
 
         self._antenna_pattern_provider = antennapattern.AntennaPatternProvider()
 
-        # initialize propagation module
-        self._prop = propagation.get_propagation_module(self._cfg['propagation']['module'])
 
-        if self._cfg['propagation']['ice_model'] == "custom":
+        if self._config['propagation']['ice_model'] == "custom":
             if ice_model is None:
                 logger.error("ice model is set to 'custom' in config file but no custom ice model is provided.")
                 raise AttributeError("ice model is set to 'custom' in config file but no custom ice model is provided.")
             self._ice = ice_model
         else:
-            self._ice = medium.get_ice_model(self._cfg['propagation']['ice_model'])
-
-        self._mout = collections.OrderedDict()
-        self._mout_groups = collections.OrderedDict()
-        self._mout_attrs = collections.OrderedDict()
+            self._ice = NuRadioMC.utilities.medium.get_ice_model(self._config['propagation']['ice_model'])
 
         # Initialize detector
         if det is None:
-            logger.status("Detectorfile {}".format(os.path.abspath(self._detectorfile)))
-            kwargs = dict(json_filename=self._detectorfile, default_station=default_detector_station,
+            logger.status("Detectorfile {}".format(os.path.abspath(detectorfile)))
+            kwargs = dict(json_filename=detectorfile, default_station=default_detector_station,
                               default_channel=default_detector_channel, antenna_by_depth=False)
             kwargs.update(det_kwargs)
             self._det = detector.Detector(**kwargs)
         else:
             self._det = det
```

Comment:
Can we break backwards compatibility? If so I am opting for removing the initialisation of the detector object from this class. The object should be passed to the class and initialized in the top level script. This also forces the user to think about it a bit more.